psycopg/psycopg/generators.py

```diff
@@ -176,7 +177,20 @@ def _fetch_many(pgconn: PGconn) -> PQGen[list[PGresult]]:
     """
     results: list[PGresult] = []
     while True:
-        res = yield from _fetch(pgconn)
+        try:
+            res = yield from _fetch(pgconn)
+        except e.DatabaseError:
+            # What might have happened here is that a previuos error disconnected
+            # the connection, for example a idle in transaction timeout.
+            # Check if we had received an error before: if that's the case
+            # just exit the loop. Our callers will handle this error and raise
+            # it as an exception.
+            for res in results:
```

Comment:
I am likely wasting your time, sorry in advance, but I don't understand how `results` is not always an empty list from line 178.

Reply:
@akrherz, the `results` list is appended below, l. 196.